src/plots/cartesian/layout_attributes.js

```diff
@@ -12,22 +12,34 @@ var constants = require('./constants');
 var HOUR = constants.HOUR_PATTERN;
 var DAY_OF_WEEK = constants.WEEKDAY_PATTERN;
 
-var tickmode = {
-    valType: 'enumerated',
-    values: ['auto', 'linear', 'array'],
-    editType: 'ticks',
-    impliedEdits: {tick0: undefined, dtick: undefined},
-    description: [
+function makeTickmode(minor) {
+    var values = ['auto', 'linear', 'array', 'sync'];
+    var description = [
         'Sets the tick mode for this axis.',
         'If *auto*, the number of ticks is set via `nticks`.',
         'If *linear*, the placement of the ticks is determined by',
         'a starting position `tick0` and a tick step `dtick`',
         '(*linear* is the default value if `tick0` and `dtick` are provided).',
         'If *array*, the placement of the ticks is set via `tickvals`',
         'and the tick text is `ticktext`.',
-        '(*array* is the default value if `tickvals` is provided).'
-    ].join(' ')
-};
+        '(*array* is the default value if `tickvals` is provided).',
+        'If *sync*, the number of ticks will sync with the overlayed axis',
+        'set by `overlaying` property.'
+    ];
+
+    if(minor) {
+        values = values.slice(0, -1);
+        description = description.slice(0, -2);
+    }
+
+    return {
+        valType: 'enumerated',
+        values: values,
+        editType: 'ticks',
+        impliedEdits: {tick0: undefined, dtick: undefined},
+        description: description.join(' ')
+    };
+}
```

Comment:
Maybe not a big deal, but this isn't going to work well with the transform that removes descriptions from the production bundle - the description will be removed from the attribute, but it'll still be in the code. Unless the minifier is smart enough to remove it as unused?

Anyway a cleaner route might be something like:
```js
var minorTickmode = {
    valType: 'enumerated',
    values: ['auto', 'linear', 'array'],
    editType: 'ticks',
    impliedEdits: {tick0: undefined, dtick: undefined},
    description: [
        'Sets the tick mode for this axis.',
        'If *auto*, the number of ticks is set via `nticks`.',
        'If *linear*, the placement of the ticks is determined by',
        'a starting position `tick0` and a tick step `dtick`',
        '(*linear* is the default value if `tick0` and `dtick` are provided).',
        'If *array*, the placement of the ticks is set via `tickvals`',
        'and the tick text is `ticktext`.',
        '(*array* is the default value if `tickvals` is provided).'
    ].join(' ')
};

var tickmode = extendFlat({}, minorTickmode, {
    values: minorTickmode.slice().concat(['sync']),
    description: [
        minorTickmode.description,
        'If *sync*, the number of ticks will sync with the overlayed axis',
        'set by `overlaying` property.'
    ].join(' ')
});
```